content/chapter-concepts/concepts-value-pairs/specifying-data-types/_index.md

```diff
@@ -43,21 +44,42 @@ The following example formats the MESSAGE field as a JSON list.
    $(format-json message=list($MESSAGE))"
 ```
 
+## Data types in {{% param "product.abbrev" %}} {#data-types}
 
 The {{% param "product.abbrev" %}} application currently supports the following data-types.
 
-  - `boolean`: Converts the data to a boolean value. Anything that begins with a `t` or `1` is converted to true, anything that begins with an `f` or `0` is converted to false.
-
-  - `datetime`: Use it only with UNIX timestamps, anything else will likely result in an error. This means that currently you can use only the `$UNIXTIME` macro for this purpose.
-
-  - `double`: A floating-point number.
-
-  - `literal`: The data as a literal string, without adding any quotes or escape characters.
-
-  - `list`: The data as a list. For details, see the list manipulation template functions in {{% xref "/chapter-manipulating-messages/customizing-message-format/reference-template-functions/_index.md" %}}.
-
-  - `int` or `int32`: 32-bit integer.
-
-  - `int64`: 64-bit integer.
-
-  - `string`: The data as a string.
+- `boolean`: Converts the data to a boolean value. Anything that begins with a `t` or `1` is converted to true, anything that begins with an `f` or `0` is converted to false.
+- `datetime`: Use it only with UNIX timestamps, anything else will likely result in an error. This means that currently you can use only the `$UNIXTIME` macro for this purpose.
+- `double`: A floating-point number.
+- `json`: A JSON snippet. (Available in {{% param "product.abbrev" %}} 4.0 and later.)
+- `literal`: The data as a literal string, without adding any quotes or escape characters.
+- `list`: The data as a list. For details, see the list manipulation template functions in {{% xref "/chapter-manipulating-messages/customizing-message-format/reference-template-functions/_index.md" %}}.
+- `null`: An unset value.
+- `int` or `int32`: 32-bit integer.
+- `int64`: 64-bit integer.
```

Comment:
At the end we decided we won't need separate int32/int64 types, so they all alias to "int". Wheather something is represented as int32 or int64 later would be determined by the destination.

There's only one case where this is important, mongodb will use int32 if the number is less than MAXINT32 and int64 otherwise.

Reply:
Maybe the new alias is integer() and not int()